test/unittest/tests/wrappers/mock_session.cpp

```diff
@@ -28,11 +28,19 @@ mock_session::~mock_session()
 {
     WT_CONNECTION_IMPL *connection_impl = _mock_connection->get_wt_connection_impl();
     if (_session_impl->block_manager != nullptr)
-        __wt_free(nullptr, _session_impl->block_manager);
+        _session_impl->block_manager_cleanup(_session_impl);
     // FIXME-WT-13505: Move terminate function to connection once circular dependency is fixed.
     if (connection_impl->file_system != nullptr)
         utils::throw_if_non_zero(connection_impl->file_system->terminate(
           connection_impl->file_system, reinterpret_cast<WT_SESSION *>(_session_impl)));
+    if (_session_impl->dhandle != nullptr) {
+        if (_session_impl->dhandle->handle != nullptr)
+            __wt_free(nullptr, _session_impl->dhandle->handle);
+        __wt_free(nullptr, _session_impl->dhandle);
+    }
+    // WiredTiger caches any allocated scratch buffer during the lifetime of the test. Free all
```

Comment:
Not blocking for merge but when do we even use the scratch buffer?

Reply:
The `__wt_block_manager_create` function uses the scratch buffer.